backtester/config/configbuilder/main.go

```diff
@@ -573,64 +551,72 @@ func addCurrencySetting(reader *bufio.Reader, usingExchangeLevelFunding bool) (*
 		if intNum > len(supported) || intNum <= 0 {
 			return nil, errors.New("unknown option")
 		}
-		setting.Asset = supported[intNum-1].String()
+		setting.Asset = supported[intNum-1]
 	}
 	for i := range supported {
 		if strings.EqualFold(response, supported[i].String()) {
-			setting.Asset = supported[i].String()
+			setting.Asset = supported[i]
 		}
 	}
 
-	var f float64
+	var d decimal.Decimal
 	fmt.Println("Enter the currency base. eg BTC")
-	setting.Base = quickParse(reader)
-	if !usingExchangeLevelFunding {
-		fmt.Println("Enter the initial base funds. eg 0")
-		parseNum := quickParse(reader)
-		if parseNum != "" {
-			f, err = strconv.ParseFloat(parseNum, 64)
-			if err != nil {
-				return nil, err
+	setting.Base = currency.NewCode(quickParse(reader))
+	if setting.Asset == asset.Spot {
+		setting.SpotDetails = &config.SpotDetails{}
```

Comment:
As this is set to be omitempty, can place the initialisation after the successful parseNum code below and then add a check at the initial quote funds parseNum block before if it's != nil to prevent this for config generation:

![image](https://user-images.githubusercontent.com/4685270/176584636-da70649d-de3e-4682-bdbe-640d9eb5eb25.png)